pkg/k8s/version/version.go

```diff
@@ -164,6 +185,72 @@ func fallbackDiscovery(client kubernetes.Interface) error {
 	return fmt.Errorf("unable to validate EndpointSlices support: %s", err)
 }
 
+func leasesFallbackDiscovery(client kubernetes.Interface) error {
+	cached.mutex.Lock()
+	// Here we check if cilium has authorization to check for leases object or not.
+	// Prior to v1.14 leases.coordination.k8s.io/v1 was not supported and hence cilium
+	// ClusterRole did not had access to corresponding API resource.
+	// Look at the helm manifest to check the ClusterRole associated with cilium Deployments
+	// corresponding to the configured Kubernetes version.
+	if cached.version.Minor < 14 && cached.version.Major == 1 {
```

Comment:
Instead of doing this, create a variable similar to the ones in line 80-82 of this file